src/requestforattestation/RequestForAttestation.ts

```diff
@@ -61,6 +63,95 @@ function getHashRoot(leaves: Uint8Array[]): Uint8Array {
   return hash(result)
 }
 
+function compressNonceAndHash(nonceHash: any[]): any {
+  if (nonceHash.length === 0) {
+    return {
+      hash: nonceHash[0],
+    }
+  }
+  return {
+    nonce: nonceHash[0],
+    hash: nonceHash[1],
+  }
+}
+
+export function compressClaimHashTree(
+  rfa: IRequestForAttestation
+): IRequestForAttestation['claimHashTree'] {
+  const result = {}
+  const claimTree = rfa.claimHashTree
+
+  Object.keys(claimTree).forEach(entryKey => {
+    result[entryKey] = Object.values(claimTree[entryKey])
+  })
+  return result
+}
+
+export function decompressClaimHashTree(
+  rfa: IRequestForAttestation[]
```

Comment:
Question:
Are you sure about the type of the function's parameter?

Reply:
No, I was going to figure out this.

Reply:
Much clearer thanks to your new naming 👍